orttraining/orttraining/python/orttraining_pybind_state.cc

```diff
@@ -162,6 +162,35 @@ struct TrainingConfigurationResult {
   optional<std::string> loss_scale_input_name;
 };
 
+#ifdef ENABLE_TRAINING_ON_DEVICE
+// Thin wrapper over internal C++ Optimizer
+struct PyOptimizer {
+  PyOptimizer(const std::string optimizer_model_uri,
+              onnxruntime::training::api::Module* model, std::vector<std::shared_ptr<IExecutionProvider>> provider) {
+    onnxruntime::SessionOptions session_option;
+    optimizer_ = std::make_unique<onnxruntime::training::api::Optimizer>(
+        optimizer_model_uri,
+        model->NamedParameters(), session_option,
+        GetTrainingORTEnv(), provider);
+  }
+
+  Status Step() const { return optimizer_->Step(); }
+  Status SetLearningRate(float lr) { return optimizer_->SetLearningRate(lr); }
+  float GetLearningRate() { return optimizer_->GetLearningRate(); }
+
+  std::shared_ptr<onnxruntime::training::api::Optimizer> GetOptimizer() { return optimizer_; }
+
+  virtual ~PyOptimizer() {}
```

Comment:
Can we avoid these by making the optimizer public. The pybind functions can just reference the optimizer and call the approriate functions.